File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/welcome_wizard/WelcomeWizardProcess.java
package com.crpr.androidcinema.domain.welcome_wizard;

import com.crpr.androidcinema.data.preferences.PreferencesService;
import com.crpr.androidcinema.domain.common.Result;

import rx.Observable;

public class WelcomeWizardProcess implements WelcomeWizard.Process{

    private final PreferencesService _prefsService;

    public WelcomeWizardProcess(PreferencesService service){
        _prefsService = service;
    }

    public Observable<Boolean> updateWelcomeWizardDone(boolean isDone){
        return Observable.just(applyWelcomeWizardIsDone(isDone));
    }

    private Boolean applyWelcomeWizardIsDone(boolean isDone){
        _prefsService.setWelcomeWizardDone(isDone);
        return true;
    }

    public Observable<WelcomeWizardResult> checkWelcomeWizardDone(){
        return Observable.just(new WelcomeWizardResult(Result.OK, _prefsService.isWelcomeWizardDone()));
    }
}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/welcome_wizard/WelcomeWizardProcess.kt
package com.crpr.androidcinema.domain.welcome_wizard

import com.crpr.androidcinema.data.preferences.PreferencesService
import com.crpr.androidcinema.domain.common.Result

import rx.Observable

class WelcomeWizardProcess(private val _prefsService: PreferencesService) : WelcomeWizard.Process {

    override fun updateWelcomeWizardDone(isDone: Boolean): Observable<Boolean> {
        return Observable.just(applyWelcomeWizardIsDone(isDone))
    }

    private fun applyWelcomeWizardIsDone(isDone: Boolean): Boolean {
        _prefsService.isWelcomeWizardDone = isDone
        return true
    }

    override fun checkWelcomeWizardDone(): Observable<WelcomeWizardResult> {
        return Observable.just(WelcomeWizardResult(Result.OK, _prefsService.isWelcomeWizardDone))
    }
}
